File: backend/java/jx/commons/sochina-commons-base/src/main/java/com/sochina/base/utils/thread/Pooled.java
package com.sochina.base.utils.thread;

public @interface Pooled {
    boolean async() default true;

    long timeout() default 500;

    String executor() default "ciToolExecutor";

    PoolOverAct poolOverAct() default PoolOverAct.REJECT;

    enum PoolOverAct {
        REJECT, BLOCK, RUN, NEW_THREAD;
    }
}

File: backend/kotlin/second/base/src/main/kotlin/com/sochina/base/utils/thread/runnable/Pooled.kt
package com.sochina.base.utils.thread.runnable

annotation class Pooled(
    val async: Boolean = true,
    val timeout: Long = 500,
    val executor: String = "ciToolExecutor",
    val poolOverAct: PoolOverAct = PoolOverAct.REJECT
) {
    enum class PoolOverAct {
        REJECT, BLOCK, RUN, NEW_THREAD
    }
}